Source/WebKit/WebProcess/GPU/media/AudioVideoRendererRemote.cpp

```diff
@@ -303,6 +303,22 @@ void AudioVideoRendererRemote::paintCurrentVideoFrameInContext(GraphicsContext&
         context.drawVideoFrame(*videoFrame, rect, ImageOrientation::Orientation::None, false);
 }
 
+RefPtr<NativeImage> AudioVideoRendererRemote::currentNativeImage() const
+{
+#if PLATFORM(COCOA)
+    RefPtr gpuProcessConnection = m_gpuProcessConnection.get();
+    RefPtr videoFrame = currentVideoFrame();
+    if (!videoFrame)
+        return nullptr;
+    ASSERT(gpuProcessConnection);
```

Comment:
I would tend to create `gpuProcessConnection` after the if check

Reply:
the aim is to move to multi-threaded implementation, by holding gpuProcessConnection we guarantee that it doesn't change as currentVideoFrame also checks that it didn't go to nullptr.